cmd/manager/main.go

```diff
@@ -32,10 +34,20 @@ func (s *server) CreateStudy(ctx context.Context, in *api_pb.CreateStudyRequest)
 	if in == nil || in.StudyConfig == nil {
 		return &api_pb.CreateStudyReply{}, errors.New("StudyConfig is missing.")
 	}
+	if in.StudyConfig.JobId != "" {
+		get_study_id, _ := dbIf.GetStudyIDfromJob(in.StudyConfig.JobId)
+		if get_study_id != "" {
+			return &api_pb.CreateStudyReply{StudyId: get_study_id}, status.Errorf(codes.AlreadyExists, "Job UUID is already registerd.")
+		}
+	}
 	studyID, err := dbIf.CreateStudy(in.StudyConfig)
 	if err != nil {
 		return &api_pb.CreateStudyReply{}, err
 	}
+	err = dbIf.RegisterStudyJob(in.StudyConfig.JobId, studyID, in.StudyConfig.JobType)
+	if err != nil {
```

Comment:
should we consider redo dbIf.CreateStudy?

Reply:
No, we shouldn't. The Reconcile() will be called asynchronously. So often a request of CreateStudy is duplicate.
This is a guard for it.

Reply:
I see. Thanks!